lutris/installer/commands.py

```diff
@@ -630,6 +630,82 @@ def autosetup_gog_game(self, file_id, silent=False):
 
+    def _get_dosbox_arguments(self, zoom_bat_path):
+        """Return the arguments for the DOSBox executable from a ZOOM Plaform installer"""
+        with open(zoom_bat_path, encoding="utf-8") as zoom_bat_file:
+            zoom_bat = zoom_bat_file.read()
+
+        # Find the line that starts with ".\DOSBOX\dosbox.exe" and extract the arguments
+        lines = zoom_bat.splitlines()
+        arguments = []
+        for line in lines:
+            if line.startswith(".\\DOSBOX\\dosbox.exe"):
+                # Extract the arguments from the line
+                arguments = line.split(" ", 1)[1].strip()
+                break
+
+        return arguments
+
+    def autosetup_zoom_platform(self, file_id, silent=False):
+        """Automatically guess the best way to install a ZOOM Platofrm game by inspecting its contents.
+        This chooses the right runner (DOSBox, Wine) for Windows game files.
+        Linux setup files don't use innosetup, they can be unzipped instead.
+        """
+        file_path = self.game_files[file_id]
+        file_list = extract.get_innoextract_list(file_path)
+        dosbox_found = False
+        scummvm_found = False
+        for filename in file_list:
+            if "dosbox.exe" in filename.lower():
+                dosbox_found = True
+            if "scummvm.exe" in filename.lower():
+                scummvm_found = True
+        if dosbox_found:
+            self._extract_innosetup(file_id)
+            dosbox_config = {}
+            single_conf = None
+            config_file = None
+            for filename in os.listdir(self.target_path):
+                if filename == "dosbox.conf":
+                    dosbox_config["main_file"] = filename
+                elif filename.endswith("_single.conf"):
+                    single_conf = filename
+                elif filename.endswith(".conf"):
+                    config_file = filename
+            if single_conf:
+                dosbox_config["main_file"] = single_conf
+            if config_file:
+                if dosbox_config.get("main_file"):
+                    dosbox_config["config_file"] = config_file
+                else:
+                    dosbox_config["main_file"] = config_file
+
+            for filename in os.listdir(self.target_path):
+                if filename.startswith("Launch") and filename.endswith(".bat"):
+                    arguments = self._get_dosbox_arguments(os.path.join(self.target_path, filename))
+
+            if len(arguments) > 0:
+                # Add the arguments to the dosbox config
+                dosbox_config["args"] = arguments
+                # Remove "config_file" from the dosbox config
+                if "config_file" in dosbox_config:
+                    del dosbox_config["config_file"]
+
+            self.installer.script["game"] = dosbox_config
+            self.installer.runner = "dosbox"
+        elif scummvm_found:
```

Comment:
Did you mean to leave the ScummVM stuff in?

Reply:
Yes, tested a with some games and the detection works fine.